### DISCUSSION TITLE
A group.next_minimized_window() function

### DISCUSSION THREAD

**Discussion by osimplex** [qtile/qtile#2562] [Show and tell]
There is in qtile the `group.next_window()` and `group.prev_window()` pair, good and working.

How can I cycle through the minimized windows only and not minimized windows only?

**Comment by ramnes** [comment]
Yes, I guess we could add the `group.{prev,next}_{minimized,maximized}_window` functions, or something like `minimized` and `maximized` as parameters to the current functions.

**Comment by arczan1** [comment]
I'm trying to add this functionality.
The problem is that the next_window() function only change focus between windows.
When you set focus to minimilized window it does not show on the screen.
Should the behaviour of the function be changed?

**Comment by osimplex** [comment]
@arczan1 the function is good as it is.
Minimized windows without a tasklist makes no sense for me. I think, maybe cicle until focus the desired window, then `lazy.window.toggle_minimize()` with one binding, or cicling will raise _all_ windows until I find that one that I want.
The problem I faced when trying to make a funcion for this was cicle only through minimized windows: first minimized window, second _etc_. I cannot go beyond the first.

**Comment by arczan1** [comment]
```
    def cmd_next_minimized_window(self):
        """
        Focus the next minimized window in group.

        Method cycles all minimized windows ing group
        """
        if not self.windows:
            return
        if not True in [window.minimized for window in self.windows]:
            return
        nxt = self.current_window
        while True:
            if self.current_window.floating:
                nxt = self.floating_layout.focus_next(nxt) or \
                    self.layout.focus_first() or \
                    self.floating_layout.focus_first(group=self)
            else:
                nxt = self.layout.focus_next(nxt) or \
                    self.floating_layout.focus_first(group=self) or \
                    self.layout.focus_first()
            if nxt.minimized:
                break
        self.focus(nxt, True)
```
I made this function but I don't know if it is good solution.
I can always add toggle_minimize() at the end.

**Comment by bodograumann** [comment]
I am trying to get this to work in my config, but get the error 

> AttributeError: 'XdgWindow' object has no attribute 'toggle_minimize'

```
@lazy.function
def unminimize_next(qtile):
    group = qtile.current_screen.group
    initial_window = group.current_window
    while True:
        if group.current_window.minimized:
            group.current_window.toggle_minimize()
            return
        qtile.current_screen.layout.next()
        if initial_window == group.current_window:
            # No minimized window found
            return
```

Any ideas?

**Reply by elParaguayo** [reply]
try `group.current_window.cmd_toggle_minimize()`

**Reply by bodograumann** [reply]
Thanks @elParaguayo ! I also had to use `qtile.current_group.cmd_next_window()`, because the layout won't cycle through minimized windows, but now it works :-)

**Comment by orjangj** [comment]
Not really sure how active this discussion is anymore, but this works quite well for me so far (somewhat similar to @bodograumann solution).

```python
@lazy.function
def unminimize_next(qtile, focus=False):
    group = qtile.current_screen.group
    for win in group.windows:
        if win.minimized:
            win.toggle_minimize()
            if focus:
                group.focus(win)
            return
```
